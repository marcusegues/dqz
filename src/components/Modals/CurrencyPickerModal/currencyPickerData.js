// @flow

export const currencyPicker: Array<{ label: string, value: string }> = [
  { label: ' ğŸ‡ªğŸ‡º EUR', value: 'EUR' },
  { label: ' ğŸ‡ºğŸ‡¸ USD', value: 'USD' },
  { label: ' ğŸ‡¦ğŸ‡ª AED', value: 'AED' },
  { label: ' ğŸ‡¦ğŸ‡± ALL', value: 'ALL' },
  { label: ' ğŸ‡¦ğŸ‡· ARS', value: 'ARS' },
  { label: ' ğŸ‡¦ğŸ‡º AUD', value: 'AUD' },
  { label: ' ğŸ‡¦ğŸ‡¿ AZN', value: 'AZN' },
  { label: ' ğŸ‡§ğŸ‡¦ BAM', value: 'BAM' },
  { label: ' ğŸ‡§ğŸ‡© BDT', value: 'BDT' },
  { label: ' ğŸ‡§ğŸ‡¬ BGN', value: 'BGN' },
  { label: ' ğŸ‡§ğŸ‡­ BHD', value: 'BHD' },
  { label: ' ğŸ‡§ğŸ‡· BRL', value: 'BRL' },
  { label: ' ğŸ‡¨ğŸ‡¦ CAD', value: 'CAD' },
  { label: ' ğŸ‡¨ğŸ‡± CLP', value: 'CLP' },
  { label: ' ğŸ‡¨ğŸ‡³ CNY', value: 'CNY' },
  { label: ' ğŸ‡¨ğŸ‡´ COP', value: 'COP' },
  { label: ' ğŸ‡¨ğŸ‡· CRC', value: 'CRC' },
  { label: ' ğŸ‡¨ğŸ‡¿ CZK', value: 'CZK' },
  { label: ' ğŸ‡©ğŸ‡° DKK', value: 'DKK' },
  { label: ' ğŸ‡©ğŸ‡´ DOP', value: 'DOP' },
  { label: ' ğŸ‡ªğŸ‡¬ EGP', value: 'EGP' },
  { label: ' ğŸ‡ªğŸ‡¹ ETB', value: 'ETB' },
  { label: ' ğŸ‡¬ğŸ‡§ GBP', value: 'GBP' },
  { label: ' ğŸ‡¬ğŸ‡¹ GTQ', value: 'GTQ' },
  { label: ' ğŸ‡­ğŸ‡° HKD', value: 'HKD' },
  { label: ' ğŸ‡­ğŸ‡³ HNL', value: 'HNL' },
  { label: ' ğŸ‡­ğŸ‡· HRK', value: 'HRK' },
  { label: ' ğŸ‡­ğŸ‡º HUF', value: 'HUF' },
  { label: ' ğŸ‡®ğŸ‡© IDR', value: 'IDR' },
  { label: ' ğŸ‡®ğŸ‡± ILS', value: 'ILS' },
  { label: ' ğŸ‡®ğŸ‡³ INR', value: 'INR' },
  { label: ' ğŸ‡®ğŸ‡¸ ISK', value: 'ISK' },
  { label: ' ğŸ‡¯ğŸ‡µ JPY', value: 'JPY' },
  { label: ' ğŸ‡°ğŸ‡ª KES', value: 'KES' },
  { label: ' ğŸ‡°ğŸ‡­ KHR', value: 'KHR' },
  { label: ' ğŸ‡°ğŸ‡· KRW', value: 'KRW' },
  { label: ' ğŸ‡°ğŸ‡¼ KWD', value: 'KWD' },
  { label: ' ğŸ‡°ğŸ‡¾ KYD', value: 'KYD' },
  { label: ' ğŸ‡°ğŸ‡¿ KZT', value: 'KZT' },
  { label: ' ğŸ‡±ğŸ‡§ LBP', value: 'LBP' },
  { label: ' ğŸ‡±ğŸ‡° LKR', value: 'LKR' },
  { label: ' ğŸ‡±ğŸ‡¾ LYD', value: 'LYD' },
  { label: ' ğŸ‡²ğŸ‡¦ MAD', value: 'MAD' },
  { label: ' ğŸ‡²ğŸ‡º MUR', value: 'MUR' },
  { label: ' ğŸ‡²ğŸ‡½ MXN', value: 'MXN' },
  { label: ' ğŸ‡²ğŸ‡¾ MYR', value: 'MYR' },
  { label: ' ğŸ‡³ğŸ‡¬ NGN', value: 'NGN' },
  { label: ' ğŸ‡³ğŸ‡´ NOK', value: 'NOK' },
  { label: ' ğŸ‡³ğŸ‡¿ NZD', value: 'NZD' },
  { label: ' ğŸ‡´ğŸ‡² OMR', value: 'OMR' },
  { label: ' ğŸ‡µğŸ‡¦ PAB', value: 'PAB' },
  { label: ' ğŸ‡µğŸ‡ª PEN', value: 'PEN' },
  { label: ' ğŸ‡µğŸ‡­ PHP', value: 'PHP' },
  { label: ' ğŸ‡µğŸ‡° PKR', value: 'PKR' },
  { label: ' ğŸ‡µğŸ‡± PLN', value: 'PLN' },
  { label: ' ğŸ‡¶ğŸ‡¦ QAR', value: 'QAR' },
  { label: ' ğŸ‡·ğŸ‡´ RON', value: 'RON' },
  { label: ' ğŸ‡·ğŸ‡¸ RSD', value: 'RSD' },
  { label: ' ğŸ‡·ğŸ‡º RUB', value: 'RUB' },
  { label: ' ğŸ‡¸ğŸ‡¦ SAR', value: 'SAR' },
  { label: ' ğŸ‡¸ğŸ‡ª SEK', value: 'SEK' },
  { label: ' ğŸ‡¸ğŸ‡¬ SGD', value: 'SGD' },
  { label: ' ğŸ‡¹ğŸ‡­ THB', value: 'THB' },
  { label: ' ğŸ‡¹ğŸ‡³ TND', value: 'TND' },
  { label: ' ğŸ‡¹ğŸ‡· TRY', value: 'TRY' },
  { label: ' ğŸ‡¹ğŸ‡¼ TWD', value: 'TWD' },
  { label: ' ğŸ‡¹ğŸ‡¿ TZS', value: 'TZS' },
  { label: ' ğŸ‡ºğŸ‡¦ UAH', value: 'UAH' },
  { label: ' ğŸ‡ºğŸ‡¾ UYU', value: 'UYU' },
  { label: ' ğŸ‡»ğŸ‡ª VEF', value: 'VEF' },
  { label: ' ğŸ‡»ğŸ‡³ VND', value: 'VND' },
  { label: ' ğŸ‡¿ğŸ‡¦ ZAR', value: 'ZAR' },
];
