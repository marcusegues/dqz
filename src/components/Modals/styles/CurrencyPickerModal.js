import { StyleSheet } from 'react-native';
import { moderateScale } from '../../../styles/Scaling';

export default StyleSheet.create({
  textInput: {
    height: 40,
    borderColor: '#A0A0A0',
    width: 110,
    borderBottomWidth: 1,
    borderBottomColor: '#9B9B9B',
    textAlign: 'center',
    marginHorizontal: 20,
    fontFamily: 'roboto_regular',
    fontSize: moderateScale(24),
    color: '#1a1a1a',
  },
  subText: {
    marginTop: 10,
    textAlign: 'center',
  },
});
