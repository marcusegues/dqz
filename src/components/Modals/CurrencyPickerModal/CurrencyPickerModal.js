// @flow
import React from 'react';
import type { ComponentType } from 'react';
// $FlowFixMe
import {
  TextInput,
  TouchableWithoutFeedback,
  View,
  Keyboard,
  Picker,
  // $FlowFixMe
} from 'react-native';

import { translate } from 'react-i18next';
import { AppModal } from '../AppModal';
import { RedButton } from '../../Buttons/RedButton';
import { pickerModalStyle } from '../styles/PickerModal';
import { PickerCard } from './subComponents/PickerCard';
import { PickerComponent } from './subComponents/PickerComponent';
import { CardHeader } from '../../QuestionAnswer/Cards/subcomponents/CardHeader';
import { CardHeaderSubText } from '../../QuestionAnswer/Cards/subcomponents/CardHeaderSubText';
import { currencyPicker } from './currencyPickerData';
import { currencyPickerModal } from '../styles/CurrencyPickerModal';
import { INDIVIDUALALLOWANCE } from '../../../model/constants';
import { currenciesArray } from '../../../model/currencies';
import type { Currency, CurrencyObject } from '../../../model/currencies';
import type { TFunction } from '../../../types/generalTypes';
import type { Amounts } from '../../../model/types/basketPeopleAmountsTypes';
import { hasOffsettingAmount } from '../../../model/utils';
import { ModalCloseText } from '../ModalCloseText';

type PickerState = {
  currency: Currency,
  amount: number,
};

type CurrencyPickerModalProps = {
  onHide: () => void,
  currencyObject: CurrencyObject,
  currencyDate: string,
  modalVisible: boolean,
  onAddAmount: (currency: Currency, amount: number) => void,
  large: boolean,
  amounts: Amounts,
  toggleModalVisible: () => void,
};

class CurrencyPickerModalInner extends React.Component<
  CurrencyPickerModalProps & { t: TFunction },
  PickerState
> {
  constructor() {
    super();
    this.state = {
      amount: 0,
      currency: 'EUR',
    };
  }

  confirmPicker() {
    const { currency, amount } = this.state;
    const { onAddAmount, onHide } = this.props;
    onAddAmount(currency, amount);
    this.setState({ amount: 0 });
    onHide();
  }

  render() {
    const {
      t,
      currencyObject,
      currencyDate,
      modalVisible,
      large,
      amounts,
      toggleModalVisible,
    } = this.props;
    const { amount, currency } = this.state;

    let disabledRedButton: boolean = typeof amount !== 'number' || amount <= 0;

    let redButtonText: string = t(['confirmPicker'], {
      value: `${currency} ${amount.toFixed(2)}`,
    });
    if (disabledRedButton) {
      redButtonText = t(['currencyPickerInvalidInput']);
    }

    if (large && currencyObject[currency] * amount < INDIVIDUALALLOWANCE) {
      disabledRedButton = true;
      redButtonText = t(['currencyPickerTooSmallLargeAmount']);
    }

    if (large && !hasOffsettingAmount(amounts, amount, currency)) {
      disabledRedButton = true;
      redButtonText = t(['currencyPickerNoOffsettingInput']);
    }

    const subButtonText: string = `CHF ${
      disabledRedButton ? '--' : (amount * currencyObject[currency]).toFixed(2)
    } (${t(['currencyPickerRate'])} ${currencyDate})`;

    let title: string = t(['currencyPickerTitle']);
    if (large) {
      title = t(['currencyPickerLargeAmountTitle'], {
        value: INDIVIDUALALLOWANCE,
      });
    }
    return (
      <AppModal
        modalVisible={modalVisible}
        onSwipe={toggleModalVisible}
        onRequestClose={toggleModalVisible}
        animationIn="slideInLeft"
        animationOut="slideOutLeft"
      >
        <PickerCard style={{ width: '100%' }}>
          <CardHeader text={title} />
          <CardHeaderSubText text={t(['currencyPickerSubTitle'])} />

          <TouchableWithoutFeedback
            onPress={Keyboard.dismiss}
            accessible={false}
          >
            <View style={pickerModalStyle.pickerContainer}>
              <PickerComponent
                selectedValue={currency}
                onValueChange={itemValue =>
                  this.setState({
                    currency: itemValue,
                  })
                }
                prompt=""
                itemStyle={{
                  fontFamily: 'roboto_medium',
                }}
                style={{ flex: 0.6 }}
              >
                {Object.keys(currencyPicker)
                  .filter(c => currenciesArray.indexOf(c) > -1)
                  .sort(
                    (a, b) =>
                      currenciesArray.indexOf(a) - currenciesArray.indexOf(b)
                  )
                  .map(c => (
                    <Picker.Item
                      key={c}
                      label={`${currencyPicker[c]} ${c}`}
                      value={c}
                    />
                  ))}
              </PickerComponent>
              <TextInput
                keyboardType="numeric"
                style={currencyPickerModal.textInput}
                onChangeText={value => this.setState({ amount: +value })}
                maxLenght="8"
                underlineColorAndroid="transparent"
                blurOnSubmit
              />
            </View>
          </TouchableWithoutFeedback>

          <View style={pickerModalStyle.redButtonWrapper}>
            <RedButton
              confirmationDisabled={disabledRedButton}
              onPress={() => {
                Keyboard.dismiss();
                this.confirmPicker();
              }}
              text={redButtonText}
            />
          </View>

          <CardHeaderSubText
            style={{
              marginTop: 10,
              textAlign: 'center',
            }}
            text={subButtonText}
          />
        </PickerCard>
        <ModalCloseText
          onModalHide={toggleModalVisible}
          text={t('closeModalText')}
        />
      </AppModal>
    );
  }
}

export const CurrencyPickerModal = (translate(['modal'])(
  CurrencyPickerModalInner
): ComponentType<CurrencyPickerModalProps>);
