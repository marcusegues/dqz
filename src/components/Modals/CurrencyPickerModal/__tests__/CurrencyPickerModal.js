import React from 'react';
import renderer from 'react-test-renderer';
import { CurrencyPickerModal } from '../CurrencyPickerModal';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

describe('Currency Picker', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<CurrencyPickerModal />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
