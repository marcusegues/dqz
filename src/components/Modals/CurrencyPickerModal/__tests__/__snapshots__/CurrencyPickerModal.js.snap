// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency Picker renders according to snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
    testID={undefined}
  />
  <View
    collapsable={undefined}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFF",
            "borderRadius": 3,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "paddingBottom": 16,
            "width": "85%",
          },
          Object {
            "elevation": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2,
          },
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "marginBottom": 31.388235294117646,
            "marginHorizontal": 31.388235294117646,
            "marginTop": 31.388235294117646,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#141414",
              "fontFamily": "roboto_regular",
              "fontSize": 24,
              "lineHeight": 62.77647058823529,
            }
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9B9B9B",
            "fontFamily": "roboto_regular",
            "fontSize": 14,
            "paddingHorizontal": 31.388235294117646,
          }
        }
      >
        
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={false}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flex": 0.25,
              },
              Object {
                "flex": 0.6,
              },
            ]
          }
        >
          <View
            style={undefined}
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "ðŸ‡ªðŸ‡º EUR",
                    "textColor": undefined,
                    "value": "EUR",
                  },
                  Object {
                    "label": "ðŸ‡ºðŸ‡¸ USD",
                    "textColor": undefined,
                    "value": "USD",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡­ CHF",
                    "textColor": undefined,
                    "value": "CHF",
                  },
                  Object {
                    "label": "ðŸ‡¦ðŸ‡ª AED",
                    "textColor": undefined,
                    "value": "AED",
                  },
                  Object {
                    "label": "ðŸ‡¦ðŸ‡± ALL",
                    "textColor": undefined,
                    "value": "ALL",
                  },
                  Object {
                    "label": "ðŸ‡¦ðŸ‡· ARS",
                    "textColor": undefined,
                    "value": "ARS",
                  },
                  Object {
                    "label": "ðŸ‡¦ðŸ‡º AUD",
                    "textColor": undefined,
                    "value": "AUD",
                  },
                  Object {
                    "label": "ðŸ‡¦ðŸ‡¿ AZN",
                    "textColor": undefined,
                    "value": "AZN",
                  },
                  Object {
                    "label": "ðŸ‡§ðŸ‡¦ BAM",
                    "textColor": undefined,
                    "value": "BAM",
                  },
                  Object {
                    "label": "ðŸ‡§ðŸ‡© BDT",
                    "textColor": undefined,
                    "value": "BDT",
                  },
                  Object {
                    "label": "ðŸ‡§ðŸ‡¬ BGN",
                    "textColor": undefined,
                    "value": "BGN",
                  },
                  Object {
                    "label": "ðŸ‡§ðŸ‡­ BHD",
                    "textColor": undefined,
                    "value": "BHD",
                  },
                  Object {
                    "label": "ðŸ‡§ðŸ‡· BRL",
                    "textColor": undefined,
                    "value": "BRL",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡¦ CAD",
                    "textColor": undefined,
                    "value": "CAD",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡± CLP",
                    "textColor": undefined,
                    "value": "CLP",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡³ CNY",
                    "textColor": undefined,
                    "value": "CNY",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡´ COP",
                    "textColor": undefined,
                    "value": "COP",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡· CRC",
                    "textColor": undefined,
                    "value": "CRC",
                  },
                  Object {
                    "label": "ðŸ‡¨ðŸ‡¿ CZK",
                    "textColor": undefined,
                    "value": "CZK",
                  },
                  Object {
                    "label": "ðŸ‡©ðŸ‡° DKK",
                    "textColor": undefined,
                    "value": "DKK",
                  },
                  Object {
                    "label": "ðŸ‡©ðŸ‡´ DOP",
                    "textColor": undefined,
                    "value": "DOP",
                  },
                  Object {
                    "label": "ðŸ‡ªðŸ‡¬ EGP",
                    "textColor": undefined,
                    "value": "EGP",
                  },
                  Object {
                    "label": "ðŸ‡ªðŸ‡¹ ETB",
                    "textColor": undefined,
                    "value": "ETB",
                  },
                  Object {
                    "label": "ðŸ‡¬ðŸ‡§ GBP",
                    "textColor": undefined,
                    "value": "GBP",
                  },
                  Object {
                    "label": "ðŸ‡¬ðŸ‡¹ GTQ",
                    "textColor": undefined,
                    "value": "GTQ",
                  },
                  Object {
                    "label": "ðŸ‡­ðŸ‡° HKD",
                    "textColor": undefined,
                    "value": "HKD",
                  },
                  Object {
                    "label": "ðŸ‡­ðŸ‡³ HNL",
                    "textColor": undefined,
                    "value": "HNL",
                  },
                  Object {
                    "label": "ðŸ‡­ðŸ‡· HRK",
                    "textColor": undefined,
                    "value": "HRK",
                  },
                  Object {
                    "label": "ðŸ‡­ðŸ‡º HUF",
                    "textColor": undefined,
                    "value": "HUF",
                  },
                  Object {
                    "label": "ðŸ‡®ðŸ‡© IDR",
                    "textColor": undefined,
                    "value": "IDR",
                  },
                  Object {
                    "label": "ðŸ‡®ðŸ‡± ILS",
                    "textColor": undefined,
                    "value": "ILS",
                  },
                  Object {
                    "label": "ðŸ‡®ðŸ‡³ INR",
                    "textColor": undefined,
                    "value": "INR",
                  },
                  Object {
                    "label": "ðŸ‡®ðŸ‡¸ ISK",
                    "textColor": undefined,
                    "value": "ISK",
                  },
                  Object {
                    "label": "ðŸ‡¯ðŸ‡µ JPY",
                    "textColor": undefined,
                    "value": "JPY",
                  },
                  Object {
                    "label": "ðŸ‡°ðŸ‡ª KES",
                    "textColor": undefined,
                    "value": "KES",
                  },
                  Object {
                    "label": "ðŸ‡°ðŸ‡­ KHR",
                    "textColor": undefined,
                    "value": "KHR",
                  },
                  Object {
                    "label": "ðŸ‡°ðŸ‡· KRW",
                    "textColor": undefined,
                    "value": "KRW",
                  },
                  Object {
                    "label": "ðŸ‡°ðŸ‡¼ KWD",
                    "textColor": undefined,
                    "value": "KWD",
                  },
                  Object {
                    "label": "ðŸ‡°ðŸ‡¾ KYD",
                    "textColor": undefined,
                    "value": "KYD",
                  },
                  Object {
                    "label": "ðŸ‡°ðŸ‡¿ KZT",
                    "textColor": undefined,
                    "value": "KZT",
                  },
                  Object {
                    "label": "ðŸ‡±ðŸ‡§ LBP",
                    "textColor": undefined,
                    "value": "LBP",
                  },
                  Object {
                    "label": "ðŸ‡±ðŸ‡° LKR",
                    "textColor": undefined,
                    "value": "LKR",
                  },
                  Object {
                    "label": "ðŸ‡±ðŸ‡¾ LYD",
                    "textColor": undefined,
                    "value": "LYD",
                  },
                  Object {
                    "label": "ðŸ‡²ðŸ‡¦ MAD",
                    "textColor": undefined,
                    "value": "MAD",
                  },
                  Object {
                    "label": "ðŸ‡²ðŸ‡º MUR",
                    "textColor": undefined,
                    "value": "MUR",
                  },
                  Object {
                    "label": "ðŸ‡²ðŸ‡½ MXN",
                    "textColor": undefined,
                    "value": "MXN",
                  },
                  Object {
                    "label": "ðŸ‡²ðŸ‡¾ MYR",
                    "textColor": undefined,
                    "value": "MYR",
                  },
                  Object {
                    "label": "ðŸ‡³ðŸ‡¬ NGN",
                    "textColor": undefined,
                    "value": "NGN",
                  },
                  Object {
                    "label": "ðŸ‡³ðŸ‡´ NOK",
                    "textColor": undefined,
                    "value": "NOK",
                  },
                  Object {
                    "label": "ðŸ‡³ðŸ‡¿ NZD",
                    "textColor": undefined,
                    "value": "NZD",
                  },
                  Object {
                    "label": "ðŸ‡´ðŸ‡² OMR",
                    "textColor": undefined,
                    "value": "OMR",
                  },
                  Object {
                    "label": "ðŸ‡µðŸ‡¦ PAB",
                    "textColor": undefined,
                    "value": "PAB",
                  },
                  Object {
                    "label": "ðŸ‡µðŸ‡ª PEN",
                    "textColor": undefined,
                    "value": "PEN",
                  },
                  Object {
                    "label": "ðŸ‡µðŸ‡­ PHP",
                    "textColor": undefined,
                    "value": "PHP",
                  },
                  Object {
                    "label": "ðŸ‡µðŸ‡­ PKR",
                    "textColor": undefined,
                    "value": "PKR",
                  },
                  Object {
                    "label": "ðŸ‡µðŸ‡± PLN",
                    "textColor": undefined,
                    "value": "PLN",
                  },
                  Object {
                    "label": "ðŸ‡¶ðŸ‡¦ QAR",
                    "textColor": undefined,
                    "value": "QAR",
                  },
                  Object {
                    "label": "ðŸ‡·ðŸ‡´ RON",
                    "textColor": undefined,
                    "value": "RON",
                  },
                  Object {
                    "label": "ðŸ‡·ðŸ‡¸ RSD",
                    "textColor": undefined,
                    "value": "RSD",
                  },
                  Object {
                    "label": "ðŸ‡·ðŸ‡º RUB",
                    "textColor": undefined,
                    "value": "RUB",
                  },
                  Object {
                    "label": "ðŸ‡¸ðŸ‡¦ SAR",
                    "textColor": undefined,
                    "value": "SAR",
                  },
                  Object {
                    "label": "ðŸ‡¸ðŸ‡ª SEK",
                    "textColor": undefined,
                    "value": "SEK",
                  },
                  Object {
                    "label": "ðŸ‡¸ðŸ‡¬ SGD",
                    "textColor": undefined,
                    "value": "SGD",
                  },
                  Object {
                    "label": "ðŸ‡¹ðŸ‡­ THB",
                    "textColor": undefined,
                    "value": "THB",
                  },
                  Object {
                    "label": "ðŸ‡¹ðŸ‡³ TND",
                    "textColor": undefined,
                    "value": "TND",
                  },
                  Object {
                    "label": "ðŸ‡¹ðŸ‡· TRY",
                    "textColor": undefined,
                    "value": "TRY",
                  },
                  Object {
                    "label": "ðŸ‡¹ðŸ‡¼ TWD",
                    "textColor": undefined,
                    "value": "TWD",
                  },
                  Object {
                    "label": "ðŸ‡¹ðŸ‡¿ TZS",
                    "textColor": undefined,
                    "value": "TZS",
                  },
                  Object {
                    "label": "ðŸ‡ºðŸ‡¦ UAH",
                    "textColor": undefined,
                    "value": "UAH",
                  },
                  Object {
                    "label": "ðŸ‡ºðŸ‡¾ UYU",
                    "textColor": undefined,
                    "value": "UYU",
                  },
                  Object {
                    "label": "ðŸ‡»ðŸ‡ª VEF",
                    "textColor": undefined,
                    "value": "VEF",
                  },
                  Object {
                    "label": "ðŸ‡»ðŸ‡³ VND",
                    "textColor": undefined,
                    "value": "VND",
                  },
                  Object {
                    "label": "ðŸ‡¿ðŸ‡¦ ZAR",
                    "textColor": undefined,
                    "value": "ZAR",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  Object {
                    "fontFamily": "roboto_medium",
                  },
                ]
              }
            />
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          blurOnSubmit={true}
          keyboardType="numeric"
          maxLenght="8"
          onChangeText={[Function]}
          style={
            Object {
              "borderBottomColor": "#9B9B9B",
              "borderBottomWidth": 1,
              "borderColor": "#A0A0A0",
              "color": "#1a1a1a",
              "fontFamily": "roboto_regular",
              "fontSize": 37.714285714285715,
              "height": 40,
              "marginHorizontal": 20,
              "textAlign": "center",
              "width": 110,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
            "marginHorizontal": 17.142857142857142,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "backgroundColor": "#DC0018",
              "borderColor": "#ddd",
              "borderRadius": 3,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 31.388235294117646,
              "width": "95%",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "lightgray",
                "borderRadius": 3,
                "flex": 1,
                "opacity": 1,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 1,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 3,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#fff",
                  "fontFamily": "roboto_medium",
                  "fontSize": 22,
                  "fontWeight": "500",
                  "paddingVertical": 31.388235294117646,
                }
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#9B9B9B",
            "fontFamily": "roboto_regular",
            "fontSize": 14,
            "marginTop": 10,
            "paddingHorizontal": 31.388235294117646,
            "textAlign": "center",
          }
        }
      >
        CHF -- ( undefined)
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "bottom": 50.285714285714285,
          "padding": 23.57142857142857,
          "position": "absolute",
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "roboto_regular",
            "fontSize": 22,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</Modal>
`;
